modal/sandbox.py

```diff
@@ -91,7 +91,36 @@ async def _get_logs():
         return data
 
 
+class _StreamWriter:
+    """Provides an interface to buffer and write logs to a sandbox stream (`stdin`)."""
+
+    def __init__(self, sandbox_id: str, client: _Client):
+        self._index = 1
+        self._sandbox_id = sandbox_id
+        self._client = client
+
+    def get_next_index(self):
+        index = self._index
+        self._index += 1
```

Comment:
This implementation means that the `Sandbox` is not `thread-safe`. If there are two instances of Sandbox performing writes, they will overwrite the `_index` here. Going to assume `Sandbox` isn't thread-safe in general now.